#! /bin/sh -e
## 62_remove_stray_mips3_in_32bit.dpatch by Ralf Baechle <ralf@linux-mips.org>
##
## DP: Remove stray .set mips3 resulting in 64-bit instruction in 32-bit kernels.
## DP: Patch author: Ralf Baechle <ralf@linux-mips.org>
## DP: Upstream status: in linux-mips git
#

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0


author	Ralf Baechle <ralf@linux-mips.org>
	Mon, 9 Jan 2006 20:09:36 +0000 (20:09 +0000)
committer	Ralf Baechle <ralf@linux-mips.org>
	Mon, 9 Jan 2006 20:09:36 +0000 (20:09 +0000)
commit	cfbc9cac62f0438cefe171736729e786b45884b8

Remove stray .set mips3 resulting in 64-bit instruction in 32-bit kernels.

Only the NMI handler was affected so this is a low impact bug.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>

file:aa18a8b7b38086e091956422a7deb0f935c6de29 -> file:1bcbd91ba1d29807c507686ba71f834799c0e0ae

diff -urN linux-mips/arch/mips/kernel/genex.S new/arch/mips/kernel/genex.S
--- linux-mips/arch/mips/kernel/genex.S	2006-01-10 11:30:59.000000000 +0000
+++ new/arch/mips/kernel/genex.S	2006-01-16 17:45:24.000000000 +0000
@@ -233,7 +233,6 @@
 NESTED(nmi_handler, PT_SIZE, sp)
 	.set	push
 	.set	noat
-	.set	mips3
 	SAVE_ALL
  	move	a0, sp
 	jal	nmi_exception_handler
